src/components/PupilViews/PupilReview/PupilReview.view.tsx

```diff
@@ -82,27 +82,35 @@ export const PupilViewsReview = (props: PupilViewsReviewProps) => {
     </OakLessonBottomNav>
   );
   const handleShareResultsClick = async () => {
+    setIsAttemptingShare(true);
     const attemptData = {
       lessonData: { slug: lessonSlug, title: lessonTitle },
       browseData: { subject: subject, yearDescription: yearDescription ?? "" },
       sectionResults: sectionResults,
     };
     try {
-      const attemptId = await logAttempt(attemptData, false);
+      const parsedAttemptData = attemptDataCamelCaseSchema.parse(attemptData);
+      const attemptId = await logAttempt(parsedAttemptData, false);
       if (!attemptId) {
         throw new Error("Failed to log attempt");
       }
+      setIsAttemptingShare(false);
       const shareUrl = `${
         process.env.NEXT_PUBLIC_CLIENT_APP_BASE_URL
       }${resolveOakHref({
-        page: "pupil-lesson-results-canonical",
+        page: "pupil-lesson-results-canonical-share",
         lessonSlug,
         attemptId,
       })}`;
-      alert("See results at " + shareUrl);
+      setTimeout(() => {
+        alert("See results at " + shareUrl);
```

Comment:
Why is this in a setTimeout ? Was this something from testing ?

Reply:
something about alerts need to be wrapped in a callback; o.w. the loading state doesnt update